PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,166 @@
+        void buildTopping();
+    }
+    class WhitePizzaBuilder : IPizzaBuilder
+    {
+        private WhitePizza pizza = new WhitePizza();
+
+        public WhitePizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+
+        }
+        public void buildDough()
+        {
+            this.pizza.add("regular white dough");
+        }
+
+        public void buildSauce()
+        {
+            this.pizza.add("Tomato sauce");
+        }
+
+        public void buildTopping()
+        {
+            this.pizza.add("Pepperoni");
+        }
+        public WhitePizza getPizza()
+        {
+            WhitePizza p = this.pizza;
+            this.Reset();
+            return p;
+        }
+    }
+    public class WhitePizza
+    {
+        private List<object> pizza = new List<object>();
+
+        public void add(string ingredient)
+        {
+            this.pizza.Add(ingredient);
+        }
+
+        public string makePizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizza.Count; i++)
+            {
+                str += " "+ this.pizza[i] + ",";
+            }
+            str = str.Remove(str.Length - 1, 1) + ".";
+            return "Pizza :" + str;
+        }
+    }
+    class WheatPizzaBuilder : IPizzaBuilder
+    {
+        private WheatPizza pizza = new WheatPizza();
+
+        public WheatPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+
+        }
+        public void buildDough()
+        {
+            this.pizza.add("thin wheat dough");
+        }
+
+        public void buildSauce()
+        {
+            this.pizza.add("Barbecue Sauce");
+        }
+
+        public void buildTopping()
+        {
+            this.pizza.add("vegies");
+        }
+        public WheatPizza getPizza()
+        {
+            WheatPizza p = this.pizza;
+            this.Reset();
+            return p;
+        }
+    }
+    public class WheatPizza
+    {
+        private List<object> pizza = new List<object>();
+
+        public void add(string ingredient)
+        {
+            this.pizza.Add(ingredient);
+        }
+
+        public string makePizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizza.Count; i++)
+            {
+                str += " " + this.pizza[i] + ",";
+            }
+            str = str.Remove(str.Length - 1, 1) + ".";
+            return "Pizza :" + str;
+        }
+    }
+
+    class Director
+    {
+        private IPizzaBuilder builder;
+        public IPizzaBuilder Builder
+        {
+            set { this.builder = value; }
+        }
+        public void makePizza()
+        {
+            this.builder.buildDough();
+            this.builder.buildSauce();
+            this.builder.buildTopping();
+        }
+    }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var waitress = new Director();
```

Comment:
Amazing Job merging the second bonus and the main project requirement into one task! 👏,
And it was a good way using a user friendly language to interact with the user 👏👏👏